internal/util/importutil/binlog_adapter.go

```diff
@@ -215,21 +219,27 @@ func (p *BinlogAdapter) Read(segmentHolder *SegmentFilesHolder) error {
 
 		// read other insert logs and use the shardList to do sharding
 		for fieldID, file := range batchFiles {
+			// outside context might be canceled(service stop, or future enhancement for canceling import task)
+			if isCanceled(p.ctx) {
+				log.Error("Binlog adapter: import task was canceled")
+				return errors.New("import task was canceled")
+			}
+
 			err = p.readInsertlog(fieldID, file, segmentsData, shardList)
 			if err != nil {
 				return err
 			}
 		}
 
-		// flush segment whose size exceed segmentSize
-		err = p.tryFlushSegments(segmentsData, false)
+		// flush segment whose size exceed blockSize
```

Comment:
flush segment for every blockSize of data read?

Reply:
For binlog import, cannot control the size of each block

Reply:
i meant that the comment needs to be updated.